sips/fork_support.md

```diff
@@ -20,56 +20,259 @@ Client implementation should calculate when a new duty starts to decide if it st
+- `Share`
+- `Config`
+- `Controller`
+
+Below, you can see how **DomainType** is linked with the main structures.
+
+<p align="center">
+<img src="./images/fork_support/domain_type_presence.png"  width="30%" height="10%">
+</p>
+
+### Use cases
+
+- When an _Instance_ needs to verify a signature. For example
+```go
+signedMessage.Signature.VerifyByOperators(signedMessage, config.GetSignatureDomainType(), types.QBFTSignatureType, operators);
+```
+
+
+- Each _Runner_, by creating the message ID when it needs to create a message. For example
+```go
+msgToBroadcast := &types.SSVMessage{
+		MsgType: types.SSVPartialSignatureMsgType,
+		MsgID:   types.NewMsgID(r.GetShare().DomainType, r.GetShare().ValidatorPubKey, r.BaseRunner.BeaconRoleType),
+		Data:    data,
+	}
+```
+
+- When the _BaseRunner_ validates a partial signature. For instance
+```go
+func (b *BaseRunner) validatePartialSigMsgForSlot(...) {
+	...
+	signedMsg.GetSignature().VerifyByOperators(signedMsg, b.Share.DomainType, types.PartialSignatureType, b.Share.Committee)
+	...
+}
+```
+
+- Indirect usage: an _Instance_ includes its _State.ID_ in messages. The _State.ID_ is initialized as the controller's _Identifier_ which contains the _DomainType_. The controller's identifier is fixed. It's used as:
+  - an argument on instance creation.
+  - to broadcast decided messages.
+  - in base validation, to verify if the message was sent to the correct controller.
+
+
+
+## Proposed Changes
+
+Along with the `DomainType`, we include a `NetworkID` type and a new `ForkData` structure.
+
+
+```go
+// NetworkID are intended to separate different SSV networks. A network can have many forks in it.
+type NetworkID byte
 
 const (
-    GenesisChain = 0x0
-    PrimusChain  = 0x1
-    ShifuChain   = 0x2
-    JatoChain    = 0x3
+	MainnetNetworkID = NetworkID(0x0)
+	PrimusNetworkID  = NetworkID(0x1)
+	ShifuNetworkID   = NetworkID(0x2)
+	JatoNetworkID    = NetworkID(0x3)
 )
 
+// DomainType is a unique identifier for signatures, 2 identical pieces of data signed with different domains will result in different sigs
+type DomainType [4]byte
+
+// DomainTypes represent specific forks for specific chains, messages are signed with the domain type making 2 messages from different domains incompatible
 var (
-    GenesisMainnet = DomainType{0x0, 0x0, GenesisChain, 0x0}
-    PrimusTestnet  = DomainType{0x0, 0x0, PrimusChain, 0x0}
-    ShifuTestnet   = DomainType{0x0, 0x0, ShifuChain, 0x0}
-    ShifuV2Testnet = DomainType{0x0, 0x0, ShifuChain, 0x1}
-    JatoTestnet    = DomainType{0x0, 0x0, JatoChain, 0x1}
+	GenesisMainnet = DomainType{0x0, 0x0, byte(MainnetNetworkID), 0x0}
+	PrimusTestnet  = DomainType{0x0, 0x0, byte(PrimusNetworkID), 0x0}
+	ShifuTestnet   = DomainType{0x0, 0x0, byte(ShifuNetworkID), 0x0}
+	ShifuV2Testnet = DomainType{0x0, 0x0, byte(ShifuNetworkID), 0x1}
+	JatoTestnet    = DomainType{0x0, 0x0, byte(JatoNetworkID), 0x0}
+	JatoV2Testnet  = DomainType{0x0, 0x0, byte(JatoNetworkID), 0x1}
```

Comment:
Just for reference
The current(ongoing) [JatoV2 network](https://github.com/bloxapp/ssv/blob/beadd23010b4c8f8bc25cb07d00bd534a70403ef/networkconfig/jato-v2.go#L14) behave as different networkID `0x4`
 ```go
 spectypes.DomainType{0x0, 0x0, 0x4, 0x1}
 ```

Reply:
Oh okay, thanks!